Lib/test/test_type_aliases.py

```diff
@@ -168,6 +168,15 @@ def test_recursive_repr(self):
         self.assertEqual(repr(GenericRecursive[GenericRecursive[int]]),
                          "GenericRecursive[GenericRecursive[int]]")
 
+    def test_raising(self):
+        type MissingName = list[_My_X]
+        with self.assertRaisesRegex(NameError, "name '_My_X' is not defined"):
+            MissingName.__value__
```

Comment:
Let's add a test where we define `_My_X`, ensuring that the value is evaluated again, and then test that the value is cached:
```suggestion
            MissingName.__value__
        _My_X = int
        self.assertEquals(MissingName.__value__,  list[int])
        del _My_X
        self.assertEquals(MissingName.__value__,  list[int])        
```